### DISCUSSION TITLE
How to Instantly Update React Hook Form, Nested Checkbox ?

### DISCUSSION THREAD

**Discussion by muhsindeniz** [react-hook-form/react-hook-form#9483] [Q&A]
Here, after returning my Nested array with map, I added a checkbox inside. If this checkbox is false, the default input field will not be required, but if the checkbox is true, the default input field will be required. But when I click on the checkbox, the required in the fields does not update. How can I do it?

const ConfiguratorNested: React.FC<ActionParametersRendererProps> = ({
  control,
  register,
  errors,
  getValues,
  setValue,
}) => {
  const { fields, append, remove } = useFieldArray({
    control,
    name: "configuration",
  })

  return (
    <>
      <TableContainer>
        <Table variant="simple">
          <Thead>
            <Tr>
              <Th color="slate.400">Name </Th>
              <Th color="slate.400">Data Type</Th>
              <Th color="slate.400">Description </Th>
              <Th color="slate.400">Placeholder </Th>
              <Th color="slate.400">Default </Th>
              <Th color="slate.400">Required</Th>
              <Th color="slate.400"></Th>
            </Tr>
          </Thead>
          <Tbody>
            {fields.map((item, index) => {
              return (
                <Tr key={item.id} borderBottom="solid 1px" borderColor="slate.700">
                  <Td border="none">
                    <Checkbox
                      {...register(`configuration.${index}.required`)}
                      name={`configuration.${index}.required`}
                      colorScheme="green"
                      size="lg"
                    />
                  </Td>

**Comment by leapful** [comment]
@muhsindeniz 

Please provide a codesandbox url to re-produce the issue.
